File: python/datahub_mcp/hello_tool.py
from fastmcp import FastMCP

mcp = FastMCP("Demo 🚀")


@mcp.tool("Description: do a 'magic' addition between two numbers")
def magic_add(a: int, b: int) -> int:
    magic_number = 3
    return a + b + magic_number


if __name__ == "__main__":
    mcp.run(transport="http", port=8001, host="localhost")

File: java/src/main/java/com/datahub/HelloTool.java
package com.datahub;

import org.springaicommunity.mcp.annotation.McpTool;
import org.springframework.stereotype.Component;

@Component
public class HelloTool {

    @McpTool(name = "magic-add", description = "Do a 'magic' addition between two numbers")
    public int magicAdd(int a, int b) {
        int magicNumber = 3;
        return a + b + magicNumber;
    }
}
